### PR TITLE
Added page about error handling

### PR DESCRIPTION
**Pull request by igorkorsukov**
https://github.com/igorkorsukov/MuseScoreDocumentation/blob/dev/error_handling/Development/Architecture/ErrorHandling.md

### CODE REVIEW

File: Development/Architecture/ErrorHandling.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    ...
+    return make_ret(Ret::Code::Ok);
+}
+
+```cpp
+mu::Ret parse(const std::string& str)
+{
+    Ret ret = checkFormat(str);
+    if (!ret) {
+        return ret;
```
Comment (Marr11317) [inline Development/Architecture/ErrorHandling.md:44]: why return `ret` if it is a `nullptr`?

Comment (igorkorsukov) [reply Development/Architecture/ErrorHandling.md:44]: I don't understand, where does `nullptr`

We called the format check function, it returned the result, if the result is negative, we return it from our function.

Comment (Marr11317) [reply Development/Architecture/ErrorHandling.md:44]: no. if ret is 0, then return the result. It means return 0, no?

Comment (igorkorsukov) [reply Development/Architecture/ErrorHandling.md:44]: in this case, `ret` has the same behaviour as `bool`

**Pull request merged**